### PR TITLE
Run tests only if they match to the regexp

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/16664
I would like to run only [`app` test](https://github.com/giantswarm/sonobuoy-plugin/blob/master/apps_test.go) here and want to make this rule more general.

### CODE REVIEW

File: run_go_test.sh
Comment (tomahawk28) [inline run_go_test.sh:19]: if it's empty, this will assign the default value as `.*` (all tests)

Comment (rossf7) [reply run_go_test.sh:19]: Not sure about TEST_FOCUS as the name. What about TEST_PATTERN?

It kind of aligns with the docs.

```
go help test

...
'Go test' recompiles each package along with any files with names matching
the file pattern "*_test.go".
...
```

Also lets make sure we get a review from Celestial too.

Comment (tomahawk28) [reply run_go_test.sh:19]: I tried to align with the sonobuoy docs. 
```
The most common point of customization is changing the set of tests to run. This is controlled by two environment variables the test image recognizes:

E2E_FOCUS
E2E_SKIP
```
https://sonobuoy.io/docs/v0.50.0/e2eplugin/

I don't have any reason to stick with `E2E_FOCUS` so which one you prefer?

Comment (rossf7) [reply run_go_test.sh:19]: OK cool I'm fine with aligning with Sonobuoy. I wasn't used to seeing focus used in this context but it makes sense.

**Comment by MarcelMue** [comment]
Can you give a bit more explanation as to the context here?

AFAIUI sonobuoy-plugin is not needed for the apps tests? Or is it intentional to use sonobuoy for those?

**Review by rossf7 — APPROVED** [review]
LGTM when tested and Celestials 🔭 are happy :)

File: run_go_test.sh
Comment (fiunchinho) [reply run_go_test.sh:19]: I'd use the same as the `cncf` tests `E2E_FOCUS`.

**Comment by tomahawk28** [comment]
@MarcelMue We perform app tests as this is part of the sonobuoy-plugin already. You can see the actual testing here; https://github.com/giantswarm/sonobuoy-plugin/blob/master/apps_test.go
The problem was it is only included in the `/test azure` so I would like to extend this testing to AWS, KVM as well.

**Comment by fiunchinho** [comment]
> @MarcelMue We perform app tests as this is part of the sonobuoy-plugin already. You can see the actual testing here; https://github.com/giantswarm/sonobuoy-plugin/blob/master/apps_test.go
> The problem was it is only included in the `/test azure` so I would like to extend this testing to AWS, KVM as well.

You can add more pipelines to test other providers, I'm not sure I understand how these changes are related to that. We can have a video call if you want or discuss it over slack.

**Comment by tomahawk28** [comment]
I should have given a simple explanation; I would like to run only the `app` test and exclude other things (e.g. ingress, autoscaler). 
The changes are intended to run only `Test_ValidateApp` in sonobuoy-plugin. Does it make sense?

File: run_go_test.sh
Comment (fiunchinho) [inline run_go_test.sh:21]: I'm not sure this is needed. [Reading the docs](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks) I see
```
go test -run ''      # Run all tests.
```

**Comment by fiunchinho** [comment]
> I should have given a simple explanation; I would like to run only the `app` test and exclude other things (e.g. ingress, autoscaler).
> The changes are intended to run only `Test_ValidateApp` in sonobuoy-plugin.
> 
> Later, in tekton, I would like to add as below to induce the app test.
> 
> ```
>       script: |
>         #! /bin/sh
>         /sonobuoy run \
> ...
>             --plugin-env giantswarm.E2E_FOCUS=App 
> ```
> 
> Does it make sense?

Ok I think I understand now. There will be a new pipeline that we will trigger when releasing apps, that will pass the new parameter to only run the app tests.

File: run_go_test.sh
Comment (tomahawk28) [reply run_go_test.sh:21]: Good catch, I will delete the default value.

**Comment by tomahawk28** [comment]
Tested in `ghost`. 

```
sonobuoy run \
                                   --kubeconfig "cp_kubeconfig.yaml" \
                                   --namespace "vr48d-sonobuoy" \
                                   --plugin giantswarm_plugin.yaml \
                                   --plugin-env giantswarm.CP_KUBECONFIG="$(cat cp_kubeconfig.yaml)" \
                                   --plugin-env giantswarm.CLUSTER_ID="vr48d" \
                                   --plugin-env giantswarm.PROVIDER="azure" \
                                   --plugin-env giantswarm.E2E_FOCUS="App" \
                                   --mode=quick \
                                   --wait --debug
```
Result:
```
name: giantswarm
status: passed
meta:
  type: summary
items:
- name: combined-report.xml
  status: passed
  meta:
    file: results/global/combined-report.xml
    type: file
  items:
  - name: github.com/giantswarm/sonobuoy-plugin/v5
    status: passed
    items:
    - name: Test_Apps
      status: passed

```

**Comment by fiunchinho** [comment]
Before merging, can you check that the report is still correct? It should tell you the number of tests that executed, their names and so on.

**Comment by tomahawk28** [comment]
Sure, `combined-report.xml` had reported there were 1 testsuites. 
```xml
<?xml version="1.0"?>
<testsuites tests="1" errors="0" failures="0">
<testsuite tests="1" failures="0" name="github.com/giantswarm/sonobuoy-plugin/v5" time="0.099">
<properties>
<property name="go.version" value="go1.15.3"/>
</properties>
<testcase name="Test_Apps" time="0.080" classname="v5"/>
</testsuite>
</testsuites>
```

**Review by fiunchinho — APPROVED** [review]

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**
